client/my-sites/themes/controller/index.node.js

```diff
@@ -43,27 +38,8 @@ export function fetchThemeDetailsData( context, next ) {
 	}
 
 	const themeSlug = context.params.slug;
-	const theme = themeDetailsCache.get( themeSlug );
-
-	if ( theme ) {
-		debug( 'found theme!', theme.id );
-		context.store.dispatch( receiveThemeDetails( theme ) );
-		next();
-	}
 
-	themeSlug && wpcom.undocumented().themeDetails( themeSlug, ( error, data ) => {
-		if ( error ) {
-			debug( `Error fetching theme ${ themeSlug } details: `, error.message || error );
-			return;
-		}
-		const themeData = themeDetailsCache.get( themeSlug );
-		if ( ! themeData || ( Date( data.date_updated ) > Date( themeData.date_updated ) ) ) {
-			debug( 'caching', themeSlug );
-			themeDetailsCache.set( themeSlug, data );
-			context.store.dispatch( receiveThemeDetails( data ) );
-			next();
-		}
-	} );
+	themeSlug && context.store.dispatch( fetchThemeDetails( themeSlug, next ) );
```

Comment:
Does this mean that we don't move on to the render step until the API response is received?